src/cursor.c

```diff
@@ -98,13 +98,9 @@ cursor_get_from_ssd(enum ssd_part_type view_area)
 static struct wlr_surface *
 get_toplevel(struct wlr_surface *surface)
 {
-	while (surface && wlr_surface_is_xdg_surface(surface)) {
-		struct wlr_xdg_surface *xdg_surface =
-			wlr_xdg_surface_from_wlr_surface(surface);
-		if (!xdg_surface) {
-			return NULL;
-		}
-
+	struct wlr_xdg_surface *xdg_surface;
+	while (surface && (xdg_surface =
+			wlr_xdg_surface_try_from_wlr_surface(surface))) {
```

Comment:
We should likely keep an eye on this one and change it back to the while condition + assignment within the loop body when we decide to switch to 0.17.